src/main/java/seedu/rc4hdb/logic/parser/commandparsers/RemoveCommandParser.java

```diff
@@ -0,0 +1,90 @@
+package seedu.rc4hdb.logic.parser.commandparsers;
+
+import static seedu.rc4hdb.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.rc4hdb.logic.parser.CliSyntax.PREFIX_EMAIL;
+import static seedu.rc4hdb.logic.parser.CliSyntax.PREFIX_FILTER_ALL;
+import static seedu.rc4hdb.logic.parser.CliSyntax.PREFIX_FILTER_ANY;
+import static seedu.rc4hdb.logic.parser.CliSyntax.PREFIX_GENDER;
+import static seedu.rc4hdb.logic.parser.CliSyntax.PREFIX_HOUSE;
+import static seedu.rc4hdb.logic.parser.CliSyntax.PREFIX_MATRIC_NUMBER;
+import static seedu.rc4hdb.logic.parser.CliSyntax.PREFIX_NAME;
+import static seedu.rc4hdb.logic.parser.CliSyntax.PREFIX_PHONE;
+import static seedu.rc4hdb.logic.parser.CliSyntax.PREFIX_ROOM;
+import static seedu.rc4hdb.logic.parser.CliSyntax.PREFIX_TAG;
+
+import java.util.HashSet;
+
+import seedu.rc4hdb.logic.commands.modelcommands.RemoveCommand;
+import seedu.rc4hdb.logic.parser.ArgumentMultimap;
+import seedu.rc4hdb.logic.parser.ArgumentTokenizer;
+import seedu.rc4hdb.logic.parser.Parser;
+import seedu.rc4hdb.logic.parser.Specifier;
+import seedu.rc4hdb.logic.parser.exceptions.ParseException;
+import seedu.rc4hdb.model.resident.ResidentStringDescriptor;
+
+
+/**
+ * Parses input arguments and creates a new RemoveCommand object
+ */
+public class RemoveCommandParser implements Parser<RemoveCommand> {
+
+    private Specifier specifier;
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the RemoveCommand
+     * and returns a RemoveCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public RemoveCommand parse(String args) throws ParseException {
+        assert args != null : "Argument for parser is null";
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_PHONE, PREFIX_EMAIL, PREFIX_ROOM, PREFIX_GENDER,
+                        PREFIX_HOUSE, PREFIX_MATRIC_NUMBER, PREFIX_TAG, PREFIX_FILTER_ALL, PREFIX_FILTER_ANY);
+
+        if (argMultimap.getValue(PREFIX_FILTER_ANY).isPresent()) {
+            specifier = new Specifier("any");
+        }
+        if (argMultimap.getValue(PREFIX_FILTER_ALL).isPresent()) {
+            specifier = new Specifier("all");
+        }
```

Comment:
There will be a potential unintended effect, which is that due to the `argMultiMap` being able to take in both any and all prefixes, if both prefixes are included in the command, the specifier will be all.